config/default/manager/manager.yaml

```diff
@@ -21,14 +21,15 @@ data:
   RANGE_END: 02:FF:FF:FF:FF:FF
 ---
 apiVersion: apps/v1
-kind: Deployment
+kind: StatefulSet
 metadata:
   name: mac-controller-manager
   namespace: system
   labels:
     control-plane: mac-controller-manager
     controller-tools.k8s.io: "1.0"
 spec:
+  serviceName: "kubemacpool-service"
```

Comment:
is this necessary? I understand that the service is already attached by the label selector

Reply:
serviceName is mandatory for StatefulSets since apps/v1, although it is not mentioned explicitly in k8s documentation, if you dont specify it you get validation error:
`alidating data: ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec;`
This guy encountered that same error as I did:
https://github.com/kubernetes/kubernetes/issues/69608#issuecomment-590958248

Reply:
serviceName is mandatory for StatefulSets  although it is not mentioned explicitly in k8s documentation, if you dont specify it you get validation error:
alidating data: ValidationError(StatefulSet.spec): missing required field "serviceName" in io.k8s.api.apps.v1.StatefulSetSpec;